pkg/volume/gce_pd/gce_util.go

```diff
@@ -101,52 +105,95 @@ func (gceutil *GCEDiskUtil) CreateVolume(c *gcePersistentDiskProvisioner) (strin
 		case "zones":
 			zonesPresent = true
 			configuredZones = v
+		case "replica-zones":
+			replicaZonesPresent = true
+			configuredReplicaZones = v
 		case volume.VolumeParameterFSType:
 			fstype = v
 		default:
 			return "", 0, nil, "", fmt.Errorf("invalid option %q for volume plugin %s", k, c.plugin.GetPluginName())
 		}
 	}
 
-	if zonePresent && zonesPresent {
+	if zonePresent && zonesPresent && replicaZonesPresent {
+		// 111
+		return "", 0, nil, "", fmt.Errorf("a combination of zone, zones, and replica-zones StorageClass parameters must not be used at the same time")
+	} else if zonePresent && zonesPresent && !replicaZonesPresent {
+		// 110
 		return "", 0, nil, "", fmt.Errorf("both zone and zones StorageClass parameters must not be used at the same time")
+	} else if zonePresent && !zonesPresent && replicaZonesPresent {
+		// 101
+		return "", 0, nil, "", fmt.Errorf("both zone and replica-zones StorageClass parameters must not be used at the same time")
+	} else if !zonePresent && zonesPresent && replicaZonesPresent {
+		// 011
+		return "", 0, nil, "", fmt.Errorf("both zones and replica-zones StorageClass parameters must not be used at the same time")
 	}
 
 	// TODO: implement PVC.Selector parsing
 	if c.options.PVC.Spec.Selector != nil {
 		return "", 0, nil, "", fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on GCE")
 	}
 
-	var zones sets.String
-	if !zonePresent && !zonesPresent {
-		zones, err = cloud.GetAllZones()
+	if !zonePresent && !zonesPresent && replicaZonesPresent {
+		// 001 - "replica-zones" specified
+		replicaZones, err := volume.ZonesToSet(configuredReplicaZones)
 		if err != nil {
-			glog.V(2).Infof("error getting zone information from GCE: %v", err)
 			return "", 0, nil, "", err
 		}
-	}
-	if !zonePresent && zonesPresent {
-		if zones, err = volume.ZonesToSet(configuredZones); err != nil {
+
+		err = createRegionalPD(
+			name,
+			c.options.PVC.Name,
+			diskType,
+			replicaZones,
+			requestGB,
+			c.options.CloudTags,
+			cloud)
+		if err != nil {
```

Comment:
After this log a message for successfully creating a regional PD

Reply:
Logging was inside the `CreateRegionalDisk` method, but I moved it here so it's easier to follow.